orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -64,6 +75,36 @@ public T start(T execution) throws Exception {
     return execution;
   }
 
+  protected T handleStartupFailure(T execution, Throwable failure) {
+    final String canceledBy = "system";
+    final String reason = "Failed on startup: " + failure.getMessage();
+    final ExecutionStatus status = ExecutionStatus.TERMINAL;
+    executionRepository.updateStatus(execution.getId(), status);
+    Function<Execution, Execution> reloader;
+    String executionType;
+    if (execution instanceof Pipeline) {
+      executionType = "pipeline";
+      reloader = (e) -> executionRepository.retrievePipeline(e.getId());
+    } else if (execution instanceof Orchestration) {
+      executionType = "orchestration";
+      reloader = (e) -> executionRepository.retrieveOrchestration(e.getId());
+    } else {
+      log.error("Unknown execution type: " + execution.getClass().getSimpleName());
+      executionType = "unknown";
+      reloader = (e) -> {
+        e.setCancellationReason(reason);
+        e.setCanceled(true);
+        e.setCanceledBy(canceledBy);
+        e.setStatus(status);
+        return e;
+      };
```

Comment:
T handleStartupFailure(T, failure) wants to return a T that reflects the state of the T after the failure has been handled. If it was an orchestration or a pipeline, this would just reread it after calling setStatus(TERMINAL) then cancel. If T is some mystery execution type this will log an error and apply the changes to the object in memory and return that object, assuming that setStatus and cancel probably did the right thing but we don't have an api on executionRepository that is just getExecution(String id)